pkg/sql/exec_util.go

```diff
@@ -1372,6 +1373,8 @@ type ExecutorConfig struct {
 
 	// NodeDescs stores {Store,Node}Descriptors in an in-memory cache.
 	NodeDescs kvcoord.NodeDescStore
+
+	TenantCapabilitiesReader tenantcapabilities.Reader
```

Comment:
One small worry is that we are still minorly breaking the spirit of this comment on the ExecutorConfig struct:

https://github.com/ecwall/cockroach/blob/3f6f2f55cdd5663b3a31d8bb4d88a01ffc80a9dd/pkg/sql/exec_util.go#L1180-L1181

One option would be for our "default" to be a tenantcapabilities.Reader that just returns an error for both methods.

Reply:
I created a new wrapper type to return this error to avoid needing to modify the `tenantcapabilities.Reader` API.